pulsar-broker/src/main/java/org/apache/pulsar/broker/service/AbstractBaseDispatcher.java

```diff
@@ -115,6 +116,16 @@ public int filterEntriesForConsumer(Optional<MessageMetadata[]> optMetadataArray
             if (entry == null) {
                 continue;
             }
+            if (entry instanceof EntryImpl) {
+                EntryImpl entryImpl = (EntryImpl) entry;
+                if (entryImpl.skipped()) {
+                    if (entriesToFiltered == null) {
+                        entriesToFiltered = new ArrayList<>();
+                    }
+                    entriesToFiltered.add(entryImpl.getPosition());
```

Comment:
Do we need `entries.set(i, null);` and `entry.release();`?

Reply:
Fixed, PTAL, thanks! @Jason918